pkg/operators/otel-metrics/otel-metrics.go

```diff
@@ -374,6 +398,50 @@ func (mc *metricsCollector) addValHistFunc(f datasource.FieldAccessor) error {
 	switch f.Type() {
 	default:
 		return fmt.Errorf("unsupported field type for metrics value %q: %s", f.Name(), f.Type())
+	case api.ArrayOf(api.Kind_Uint32), api.ArrayOf(api.Kind_Uint64):
+		// Calc buckets
+		typeLen := 4
+		extractor := func(b []byte) uint64 { return uint64(ds.ByteOrder().Uint32(b)) }
+		if f.Type() == api.ArrayOf(api.Kind_Uint64) {
+			typeLen = 8
+			extractor = ds.ByteOrder().Uint64
+		}
+		alen := int(f.Size()) / typeLen
+
+		bucketVals := make([]int64, alen)
+		bucketValsFloat := make([]float64, alen)
+		for i := range bucketVals {
+			bucketVals[i] = int64(1 << i)
+			bucketValsFloat[i] = float64(bucketVals[i])
+		}
+
+		options = append(options, metric.WithExplicitBucketBoundaries(bucketValsFloat...))
```

Comment:
`bucketVals` are ints, but `metric.WithExplicitBucketBoundaries` expects floats. Or did I misunderstand the question?

Reply:
But it is used again a couple of lines below on the hot path, where I wanted to prevent converting back to ints:

```go
for range val {
	hist.Record(ctx, bucketVals[i], metric.WithAttributeSet(set))
}
```